src/turbomind/kernels/sampling_penalty_kernels.cu

```diff
@@ -412,6 +412,10 @@ __global__ void batchApplyRepetitionPenalty(T*           logits,
         }
     }
 
+    if (blockDim.x > 32) {
```

Comment:
I think `__syncthreads()` should be called unconditionally, or at least `__syncwarp()` when `blockDim.x < 32`.

warp may divergent here since index depends on `threadIdx.x`
```C++
if (index >= input_length && index < max_input_length) {
    continue;
}
```